torch/distributed/fsdp/wrap.py

```diff
@@ -28,6 +29,36 @@ def always_wrap_policy(*args, **kwargs) -> bool:
     """
     return True
 
+def wrap_batchnorm_individually(
+    module: nn.Module,
+    recurse: bool,
+    *args,
+    **kwargs,
+) -> bool:
+    """
+    A policy that wraps ``BatchNorm`` instances in their own FSDP unit.
+    """
+    if recurse:
+        # always recurse
+        return True
+    # if not recursing, decide whether we should wrap based on whether it is a
+    # BN layer or not.
+    return isinstance(module, _BatchNorm)
+
+def or_policy(
+    module: nn.Module,
+    recurse: bool,
+    unwrapped_params: int,
+    policies,
+) -> bool:
+    """
+    A policy that wraps ``module`` if any policy in the passed in iterable of
+    ``policies`` returns ``True``.
+    """
+    return any(
+        policy(module, recurse, unwrapped_params) for policy in policies
```

Comment:
I see, this policy combined with wrap_batchnorm_individually makes sure the wrapping decision is completely bottom to up, wrap_batchnorm_individually only takes care of batchnorm leaf module, user specified policy will make decision when returns from bottom to up in the tree traversal